x-pack/plugins/infra/server/lib/alerting/log_threshold/log_threshold_executor.ts

```diff
@@ -247,11 +258,12 @@ const getESQuery = (
   alertParams: Omit<RuleParams, 'criteria'> & { criteria: CountCriteria },
   timestampField: string,
   indexPattern: string,
-  runtimeMappings: estypes.MappingRuntimeFields
+  runtimeMappings: estypes.MappingRuntimeFields,
+  timestamp: number
 ) => {
   return hasGroupBy(alertParams)
-    ? getGroupedESQuery(alertParams, timestampField, indexPattern, runtimeMappings)
-    : getUngroupedESQuery(alertParams, timestampField, indexPattern, runtimeMappings);
+    ? getGroupedESQuery(alertParams, timestampField, indexPattern, runtimeMappings, timestamp)
```

Comment:
Out of curiosity, does it mean now we will have different results after this update? Do we know the gain for accuracy we gained in terms of ms?

Reply:
Since the timestamp is now determined by the alerting earlier in the call stack I would expect the time interval this es query filters for would shift a few milliseconds earlier. The size of the interval would still remain the same.

Reply:
This shouldn't really affect the accuracy as much as ensure the timestamps used for the alert is consistent with the timestamp used for the query. Before this change, the query timestamp could differ from the reported timestamp in `scheduleActions`; depending on how much time has passed between when we ran the query and when we scheduled the action.

The ultimate result is that the definition of "NOW" is aligned with `startedAt` which will be consistent with the event log captured by the Alerting framework. It also makes these functions testable since we can control the timestamp instead of relying on `Date.now()` which obviously changes on every execution.